Meta/CMake/common_compile_options.cmake

```diff
@@ -65,7 +65,10 @@ elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64")
     # FIXME: Remove this branch once -march=native is supported.
 else()
     # In all other cases, compile for the native architecture of the host system.
-    add_cxx_compile_options(-march=native)
+    # # This is only valid when not cross-compiling.
+    if (NOT CMAKE_CROSSCOMPILING)
```

Comment:
Maybe hoist this up to the `else()` and make that an `elseif (NOT CMAKE_CROSSCOMPILING)`?